### PR TITLE
feat: CI-9438: passing github action with as individual vars

### PR DESCRIPTION
**Pull request by devkimittal**

### CODE REVIEW

**Review by raghavharness — APPROVED** [review]

File: plugin/github/env.go [hunk 1]
```diff
@@ -100,15 +100,30 @@ func parseOwner(s string) (owner string) {
...
+		if val, ok := envVars["PLUGIN_WITH"]; ok {
+			with, err := strToMap(val)
+			if err != nil {
+				return nil, errors.Wrap(err, "with attribute is not of map type with key & value as string")
+			}
+			return with, nil
 		}
-
-		return with, nil
+		return nil, nil
```
Comment (vistaarjuneja) [inline plugin/github/env.go]: can remove this if already returning at L113

File: plugin/github/env.go [hunk 2]
```diff
@@ -100,15 +100,30 @@ func parseOwner(s string) (owner string) {
...
+	} else {
+		m := make(map[string]string)
+		prefix := "PLUGIN_WITH_"
```
Comment (vistaarjuneja) [inline plugin/github/env.go]: keep `PLUGIN_WITH` in a single place at top of the function. And prefix check could be just for `PLUGIN_WITH`. You could reuse the string everywhere in this function.
`pluginWithVar := "PLUGIN_WITH"`

**Review by vistaarjuneja — APPROVED** [review]

File: plugin/github/env.go (hunk 2 above)
Comment (devkimittal) [reply plugin/github/env.go]: I'm passing PLUGIN_WITH_<varname> from ci manager, so we should be checking for prefix PLUGIN_WITH_

File: plugin/github/env.go [hunk 3]
```diff
@@ -100,15 +100,30 @@ func parseOwner(s string) (owner string) {
 }
 
 func getWith(envVars map[string]string) (map[string]string, error) {
-	if val, ok := envVars["PLUGIN_WITH"]; ok {
+	PLUGIN_WITH := "PLUGIN_WITH"
```
Comment (vistaarjuneja) [inline plugin/github/env.go:103]: Nit: golang variables should be named in camel case unless declaring constants

**Review by vistaarjuneja — APPROVED** [review]

File: plugin/github/env.go (hunk 3 above)
Comment (devkimittal) [reply plugin/github/env.go:103]: will keep it in mind.

**Pull request merged**